app/scripts/app.js

```diff
@@ -42,6 +47,23 @@ angular.module('yourStlCourts').config(function($stateProvider, $urlRouterProvid
         }
       }
     })
+    .state('courtSearchInfo', {
+      url: '/courts/{courtId}',
+      templateUrl: 'views/courtSearchInfo.html',
+      controller: 'CourtSearchInfoCtrl as ctrl',
+      params: {
+        court: {value: undefined}
+      },
+      resolve: {
+        court: function ($stateParams,Courts) {
+          if ($stateParams.courtId != ""){
```

Comment:
empty string doesn't seem appropriate. If it's not provided, wouldn't it be undefined or null? Plus, since the id is provided in the URL, the state doesn't get even hit if there is no courtId. So this empty string shouldn't even be possible.